lib/hashie/extensions/coercion/active_model.rb

```diff
@@ -0,0 +1,85 @@
+require 'active_model/type'
+
+module Hashie
+  module Extensions
+    module Coercion
+      module ActiveModel
+        # Symbol is the one glaring type omission. Define it if it is not
+        # already defined.
+        begin
+          ::ActiveModel::Type.lookup(:symbol)
+        rescue ArgumentError
+          class Symbol < ::ActiveModel::Type::String
+            def type
+              :symbol
+            end
+
+            private
+
+            # The symbol type leverages in coercion logic in the String
+            # superclass and simply uses `to_sym` to covert to a symbol.
+            def cast_value(value)
+              super.to_sym
+            end
+          end
+          ::ActiveModel::Type.register(:symbol, Symbol)
+        end
+
+        # Array of symbols of the included ActiveModel types.
+        ACTIVE_MODEL_TYPES = ::ActiveModel::Type.registry
```

Comment:
Reformatting this to look like the following can help with readability:

```ruby
ACTIVE_MODEL_TYPES =
  ::ActiveModel::Type
    .registry
    .send(:registrations)
    .map { |r| r.send(:name) }
    .freeze
```

Also, please use `#__send__` instead of `#send`. Since we're a library, we have to gracefully handle people doing strange things like overriding the `#send` method.

Lastly, why `.send(:registrations)` and not just `.registrations`?